scripts/nbconverted/explore_relationship_AandB.py

```diff
@@ -0,0 +1,585 @@
+                else 'threshold' if 0.49<= test_samples_sorted.loc[x,rep_gene_A]<=0.51
+                else 'greater',
+                samples
+            )      
+        )
+    )
+)
+data_encoded_labeled.head()
+
+
+# In[19]:
+
+
+# Heatmap sorted by gene expression signature
+# colormap: 
+#     offset - red
+#     less - blue
+#     threshold - green
+#     greater- magenta
+threshold_groups = data_encoded_labeled["threshold_group"]
+lut = dict(zip(threshold_groups.unique(), "rbgm"))
+row_colors = threshold_groups.map(lut)
+
+sns.clustermap(data_encoded,
+               row_cluster=True,
+               col_cluster=False,
+               metric="correlation",
+               row_colors=row_colors,
+               figsize=(50,50))
+
+
+# ### Collapse gene expression pattern
+# It is difficult to see any trends in large heatmap so collapse the encoded gene expression using mean per group
+
+# In[20]:
+
+
+# Get mean of samples in each group ()
+
+data_encoded_mean_less = pd.DataFrame(data_encoded_labeled[data_encoded_labeled.threshold_group == 'less'].mean(numeric_only=True))
+data_encoded_mean_threshold = pd.DataFrame(data_encoded_labeled[data_encoded_labeled.threshold_group == 'threshold'].mean(numeric_only=True))
+data_encoded_mean_greater = pd.DataFrame(data_encoded_labeled[data_encoded_labeled.threshold_group == 'greater'].mean(numeric_only=True))
+
+
+# In[21]:
+
+
+# Plot mean expression for each group
+plt.figure(figsize=(100, 10))
+sns.heatmap(data_encoded_mean_less.T, annot = True, cmap = "RdBu_r")
+plt.figure(figsize=(100, 10))
+sns.heatmap(data_encoded_mean_threshold.T, annot = True, cmap = "RdBu_r")
+plt.figure(figsize=(100, 10))
+sns.heatmap(data_encoded_mean_greater.T, annot = True, cmap = "RdBu_r")
+
+
+# ### Which features have the largest difference as we cross the threshold?
+
+# In[22]:
+
+
+# Difference in means
+diff_data_encoded = data_encoded_mean_greater - data_encoded_mean_less
+abs_diff_data_encoded = abs(diff_data_encoded)
+
+# Get top 5 features
+top_features = abs_diff_data_encoded[0].nlargest()
+top_features
+
+
+# In[23]:
+
+
+# Check sign of top features
+feature_names = [int(l) for l in top_features.index]
+diff_data_encoded.iloc[feature_names]
+
+
+# In[24]:
+
+
+# What is the weight of the offset vector for this top feature?
+top_feature = top_features.index[2]
+print(offset_vae_space[top_feature])
+
+sns.distplot(offset_vae_space)
+
+
+# ## What is the trend of latent features across the threshold?
+# 
+# Plot trend of encoded values along threshold for top feature
+
+# In[25]:
+
+
+# Join sorted original expression of A and encoded expression of top feature
+original_A_exp = test_samples_sorted[rep_gene_A]
+feature_values = data_encoded[top_feature]
+
+trend_feature_df = pd.merge(original_A_exp.to_frame('gene A untransformed'),
+                      feature_values.to_frame('values of feature {}'.format(top_feature)),
+                      left_index=True, right_index=True)
+trend_feature_df.head()
```

Comment:
How do you go from your top feature being 19 to 106? Is it because 19 is too negative?

Reply:
There are 5 top features that showed the largest ABSOLUTE difference (regardless of sign).  The top feature is 19.  106 is the 3rd most different feature.  I was manually playing around with the values by changing the index in ```top_feature = top_features.index[2]```